weaver/wps_restapi/swagger_definitions.py

```diff
@@ -3935,6 +3936,70 @@ class ExecuteCollectionInput(FilterSchema, SortBySchema, PermissiveMappingSchema
     )
 
 
+class ExecuteNestedProcessReference(ExtendedMappingSchema):
+    title = "ExecuteNestedProcessReference"
+    # 'process' is required for a nested definition, otherwise it will not even be detected as one!
+    process = ProcessURL(description="Process reference to be executed.")
+
+
+class ExecuteNestedProcessParameters(ExtendedMappingSchema):
+    """
+    Dynamically defines the nested process parameters with recursive schema handling.
+
+    This class must create the nested properties dynamically because the required classes are not yet defined, and
+    those required definitions also depend on this class to define the nested process as a possible input value.
+
+    .. seealso::
+        - https://docs.pylonsproject.org/projects/colander/en/latest/binding.html
+    """
+    title = "ExecuteNestedProcessParameters"
+    _sort_first = ["process", "inputs", "outputs", "properties", "mode", "response"]
+    _schema_extra = {
+        "type": null,
+        "title": "ExecuteNestedProcessParameters",
+        "$ref": f"{OAS3DefinitionHandler.json_pointer}ExecuteProcessParameters"
+    }
+
+    @colander.deferred
+    def _children(self, __bindings):
+        # type: (Dict[str, Any]) -> List[colander.SchemaNode]
+        self.children = [node.clone() for node in ExecuteProcessParameters().children]
```

Comment:
Isn't confusing calling it a bind if it's not a shallow copy ?
Shouldn't it be` self.children = list(ExecuteProcessParameters().children)`

Reply:
Limitation from the parent `colander` library that called the method `bind`. Nested nodes have to be cloned, otherwise you could have by-ref conflicting updates if similar definitions are used simultaneously.

Reply:
I've added a test in [`ef93c74` (#748)](https://github.com/crim-ca/weaver/pull/748/commits/ef93c7406bf3ef18a997a642824f5c14fdc6bc65) so you can get a better understanding of the feature and its behavior.